src/lib.rs

```diff
@@ -206,6 +206,7 @@ impl<T> Counter for T
 /// = 2048 * 2^(k-1)`, which is the k-1'th bucket's end. So, we would use the previous bucket
 /// for those lower values as it has better precision.
 ///
+#[derive(Debug)]
```

Comment:
I did see the string matches, but I'd much prefer it if the library instead returned an `enum Error` where the strings we currently return in `Err` were instead tagged to the appropriate enum variants.

That all said, I guess I'd be okay with accepting a `derive(Debug)` and then have a later PR implement it more sanely. In particular, I think the output of `counts` is just wholly unhelpful. The same goes for having an `Error` enum; we could do that at a later time instead.